devices/power.go

```
@@ -33,28 +42,26 @@ func amtPowerWrite(device *sdk.Device, data *sdk.WriteData) error {
 	// When writing to a AMT Power device, we always expect there to be
 	// raw data specified. If there isn't, we return an error.
 	if len(raw) == 0 {
-		return fmt.Errorf("no values specified for 'raw', but required")
+		logger.Error("no values specified for 'raw', but required")
+		return nil
 	}
 
 	if action == "state" {
-		cmd := string(raw[0])
-
-		switch strings.ToLower(cmd) {
-		case "on":
-			logger.Debug("AMT Power On")
-		case "off":
-			logger.Debug("AMT Power Off")
-		case "reset":
-			logger.Debug("AMT Power Resetting")
-		case "cycle":
-			logger.Debug("AMT Power Cycling")
-		default:
-			return fmt.Errorf("unsupported command for amt power 'state' action: %s", cmd)
+		commandName := string(raw[0])
+		cmd := exec.Command("python", "scripts/power.py", device.Data["ip"], // nolint: gas
+			device.Data["password"], commandName)
+
+		_, err := cmd.Output()
+
+		if err != nil {
+			logger.Errorf("Error: %s\n", cmd.Stderr)
+			return err
 		}
 
 	} else {
 		// If we reach here, then the specified action is not supported.
-		return fmt.Errorf("action '%s' is not supported for AMT power devices", action)
+		logger.Error("action '%s' is not supported for AMT power devices", action)
+		return nil
```

revert, we want to return the error, not nil